### ISSUE TITLE
New Series types: Science Charts

### ISSUE THREAD

**Issue by Izothep** [highcharts/highcharts#12531]
There is a possibility to create new series types for Scientific data visualisation, such as:

- [ ]  Histogram 3D
- [ ] Column 3D with full suport for z axis
- [ ] Contour
- [ ] Vector fields
- [ ] surf/ribbon plots
- [ ] Mesh
- [ ] Streamline
- [ ] Ternary plots
- [ ] Electric circuit diagrams

**Comment by cnighswonger** [comment]
@pawelfus Smith charts would also be a nice addition falling into this category.

**Comment by pawelfus** [comment]
Added, thanks!

**Comment by karolkolodziej** [comment]
unstale

**Comment by SteveF311** [comment]
Please add contour plot support to Highcharts.

https://www.statisticshowto.com/contour-plots/

Paulo did a great job on his below plug-in but have to hack the heatmap chart to make it work with the Highcharts C# wrapper. I would argue contour plots are the highest priority on the above list (they have the most use cases and are an extension of existing heatmap; also they are certainly more useful than 3D histograms and columns which are typically just chartjunk).

https://github.com/paulo-raca/highcharts-contour/?tab=readme-ov-file

**Comment by ddragula** [comment]
Hi @SteveF311, I agree that contour maps from this list seem to be the most useful. I would like to ask what data format you would like to be able to generate such a map from? Defining the coordinates of the isolines or maybe extracting it from discrete data using e.g. the Marching Squares algorithm, or even something else?

**Comment by SteveF311** [comment]
Hi @ddragula , I may not understand your question, but the current usage is based on this plug-in (https://github.com/paulo-raca/highcharts-contour/?tab=readme-ov-file) and the data is XYZ. XY for the spatial coordinates and then Z for the "elevation" coordinate or value at said XY coordinate. Preference is to automatically determine the contour lines (isolines). Paulo's plug-in uses Delaunay triangulation, which works just fine. Here is a below C# example. We have to hack the Highcharts output to get the correct syntax for Paulo's plug-in to work (e.g., replace heatmap with contour, add the checkbox, etc). This request is similar to Highcharts heatmap with the ColorAxis Stops interpolation, but explicitly from a contour map perspective as a built-in chart type to both Highcharts JS and C# wrapper.

highcharts = new Highcharts
{
	ID = Container,
	Series = series,
	Chart = new Chart
	{
		Type = ChartType.Heatmap,
		AlignThresholds = true,
		ZoomType = ChartZoomType.Xy,
	},
	ColorAxis = new ColorAxis
	{
		Type = "linear",
		Stops = new List<Stop>()
		{
			new Stop() { Color = ( "#2124CF"), Position = 0.0 },
			new Stop() { Color = "#fffbbc", Position = 0.5 },
			new Stop() { Color = ( "#FF8F00"), Position = 1.0 }
		},
		Min = MinZ,
		Max = MaxZ,
		TickInterval = TickIntervalZ,
		StartOnTick = true,
		EndOnTick = true
	},
	XAxis = new List<XAxis>
	{
		new XAxis
		{
			Type = "linear",
			Title = new XAxisTitle
			{
				Text = "Measurement X"
			},
			CrosshairBool = true,
			GridZIndex = 0,
			GridLineWidth = 1,
			Reversed = false
		}
	},
	YAxis = new List<YAxis>
	{
		new YAxis
		{
			Type = "linear",
			Title = new YAxisTitle
			{
				Text = "Measurement Y"
			},
			CrosshairBool = true,
			GridZIndex = 0,
			GridLineWidth = 1,
			Reversed = false
		}
	},
	PlotOptions = new PlotOptions
	{
		Heatmap = new PlotOptionsHeatmap
		{
			AllowPointSelect = true,
			EnableMouseTracking = true
		},
		Scatter = new PlotOptionsScatter
		{
			Marker = new PlotOptionsScatterMarker
			{
				Symbol = "circle"
			},
			AllowPointSelect = false,
			EnableMouseTracking = false,
			ShowInLegend = false
		}
	}
};

**Comment by cascading-jox** [comment]
> Hi [@SteveF311](https://github.com/SteveF311), I agree that contour maps from this list seem to be the most useful. I would like to ask what data format you would like to be able to generate such a map from? Defining the coordinates of the isolines or maybe extracting it from discrete data using e.g. the Marching Squares algorithm, or even something else?

For the [use case](https://github.com/highcharts/highcharts/issues/17324) in my application I would like to just enter the data points and have the library calculate the lines using something like the Marching Squares algorithm.

To give you more context about my specific use case - I'm working with temperature data on a regular grid (hourly readings over multiple years, like the El Bosque chart above). So my data is basically:
- Grid points with temperature values
- X axis: hours (0-23)
- Y axis: months/dates

What would be really useful:
- Being able to specify which contour levels I want (e.g. every 5°C)
- Smooth color interpolation between data points (not discrete cells like a traditional heatmap)
- The contour lines drawn on top of this continuous color surface

Basically, I'd love to just be able to pass in my temperature grid and get something like the chart in the image – a smooth interpolated color surface with contour lines.

There are some good reference implementations like [d3-contour](https://github.com/d3/d3-contour) or [conrec.js](https://github.com/jasondavies/conrec.js).

**Comment by ddragula** [comment]
Hi @cascading-jox, thanks for the details! Some time ago I made a [PoC](https://github.com/ddragula/webgpu-ts-tests/tree/demo/contour-chart) for a potential Highcharts module ([sample](https://ddragula.github.io/webgpu-ts-tests/contour-charts/)), how it could work on Delaunay Triangulation, rendering a heatmap using WebGPU - you might find it useful before it's implemented in Highcharts.
